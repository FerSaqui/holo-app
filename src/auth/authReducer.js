/**
 * En un reducer, la acciÃ³n va a modificar el state y react va a renderiza/actualizar lo que tenga que actualizar
 */

import { types } from "../types/types";

export const authReducer = (state = {}, action) => {

    switch ( action.type ) {
        case types.login:
            return {
                ...action.payload,
                logged: true
            }

        case types.logout:
            return {
                logged: false
            }
    
        default:
            return state;
    }

}