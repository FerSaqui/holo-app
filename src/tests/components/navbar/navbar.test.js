import { mount} from "enzyme";
import { MemoryRouter, Route, Routes } from "react-router-dom";
import { AuthContext } from "../../../auth/autContext";
import { Navbar } from "../../../components/Navbar/Navbar";
import { types } from "../../../types/types";

const dispatch = jest.fn();

/**Simular el hook o la librería de react-router-dom */
const mockNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useNavigate: () => mockNavigate
}));

/*****************************************************/

describe('Pruebas en <navbar />', () => {
    const contextValue = {
        user : {
            logged: true,
            name: "Fernando"
        },
        dispatch
    }

    test('Debe de mostrarse correctamente', () => {

        // const wrapper = mount(
        //     <AuthContext.Provider value={ contextValue }>
        //         <MemoryRouter>
        //             <Navbar />
        //         </MemoryRouter>
        //     </AuthContext.Provider>
        // );
        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <MemoryRouter initialEntries={ ["/"] }>
                    <Routes>
                        <Route path="/" element= {<Navbar />} />
                    </Routes>
                </MemoryRouter>
            </AuthContext.Provider>
        );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find("#txtNombre").text().trim() ).toBe(contextValue.user.name);
    });

    test('Debe de llamar el logout, llamar navigate y el dispatch con los argumentos', () => {
        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <MemoryRouter>
                    <Navbar />
                </MemoryRouter>
            </AuthContext.Provider>
        );

        // wrapper.find(".bg-danger").simulate("click");
        wrapper.find(".bg-danger").prop("onClick")(); //Alternativa a simular el click al botón
        expect(contextValue.dispatch).toHaveBeenCalledWith({type: types.logout});
        expect(mockNavigate).toHaveBeenCalledWith("/login", {replace: true});
    });

});