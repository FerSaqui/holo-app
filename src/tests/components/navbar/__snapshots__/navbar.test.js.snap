// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <navbar /> Debe de mostrarse correctamente 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark"
>
  <div
    className="container-fluid"
  >
    <a
      aria-current="page"
      className="navbar-brand active"
      href="/"
      onClick={[Function]}
    >
      Hololive
    </a>
    <button
      aria-controls="navbarText"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-bs-target="#navbarText"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarText"
    >
      <ul
        className="navbar-nav me-auto mb-2 mb-lg-0"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/holoEN"
            onClick={[Function]}
          >
            Holo EN
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/holoJP"
            onClick={[Function]}
          >
            Holo JP
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/search"
            onClick={[Function]}
          >
            Search
          </a>
        </li>
      </ul>
      <ul
        className="navbar-nav"
      >
        <span
          className="nav-item text-white ms-5"
          id="txtNombre"
        >
          Fernando
        </span>
        <button
          className="nav-item bg-danger"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </div>
</nav>
`;
